feature/experimental/bgp/ate_tests/bgp_remove_private_as/bgp_remove_private_as_test.go

```diff
@@ -0,0 +1,347 @@
+	dutAS                  = 500
+	ateAS1                 = 100
+	ateAS2                 = 200
+	plenIPv4               = 30
+	plenIPv6               = 126
+	removeASPath           = true
+)
+
+var (
+	dutSrc = attrs.Attributes{
+		Desc:    "DUT to ATE source",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::192:0:2:1",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	ateSrc = attrs.Attributes{
+		Name:    "ateSrc",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::192:0:2:2",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	dutDst = attrs.Attributes{
+		Desc:    "DUT to ATE destination",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::192:0:2:5",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	ateDst = attrs.Attributes{
+		Name:    "atedst",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::192:0:2:6",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+)
+
+// configureDUT configures all the interfaces on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	dc := gnmi.OC()
+	i1 := dutSrc.NewOCInterface(dut.Port(t, "port1").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	i2 := dutDst.NewOCInterface(dut.Port(t, "port2").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i2.GetName()).Config(), i2)
+}
+
+// verifyPortsUp asserts that each port on the device is operating.
+func verifyPortsUp(t *testing.T, dev *ondatra.Device) {
+	t.Helper()
+	for _, p := range dev.Ports() {
+		status := gnmi.Get(t, dev, gnmi.OC().Interface(p.Name()).OperStatus().State())
+		if want := oc.Interface_OperStatus_UP; status != want {
+			t.Errorf("%s Status: got %v, want %v", p, status, want)
+		}
+	}
+}
+
+// bgpCreateNbr creates a BGP object with neighbors pointing to ateSrc and ateDst.
+func bgpCreateNbr(localAs, peerAs uint32, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	nbr1v4 := &bgpNeighbor{as: ateAS1, neighborip: ateSrc.IPv4, isV4: true, peerGrp: peerGrpName1}
+	nbr2v4 := &bgpNeighbor{as: ateAS2, neighborip: ateDst.IPv4, isV4: true, peerGrp: peerGrpName2}
+	nbrs := []*bgpNeighbor{nbr1v4, nbr2v4}
+
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+	global := bgp.GetOrCreateGlobal()
+	global.RouterId = ygot.String(dutDst.IPv4)
+	global.As = ygot.Uint32(localAs)
+	global.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+
+	// Note: we have to define the peer group even if we aren't setting any policy because it's
+	// invalid OC for the neighbor to be part of a peer group that doesn't exist.
+	pg1 := bgp.GetOrCreatePeerGroup(peerGrpName1)
+	pg1.PeerAs = ygot.Uint32(ateAS1)
+	pg1.PeerGroupName = ygot.String(peerGrpName1)
+
+	pg2 := bgp.GetOrCreatePeerGroup(peerGrpName2)
+	pg2.PeerAs = ygot.Uint32(ateAS2)
+	pg2.PeerGroupName = ygot.String(peerGrpName2)
+
+	for _, nbr := range nbrs {
```

Comment:
For eBGP neighbors , there needs to be a Allow/Permit Route-policy under the Peer-group AFi/SAFI. Could you please add them as default , and use deviation RoutePolicyUnderAFIUnsupported for the config without RPL ?

Reply:
@arvbaska1 , we are not enabling any route policy in this test, did not get why we need to enable it under AFI-SAFI.

Reply:
As Per RFC8212 for eBGP a route-policy is a must to allow routes to be advertised/received, so we need to add an Allow/Permit policy. RPL under the Peer-group AFI/SAFI is supposed to be the default , similar to changes in PR#1538/1679.

Reply:
updated, please check.